<Type Name="XamlInternalException" FullName="System.Xaml.XamlInternalException">
  <Metadata><Meta Name="ms.openlocfilehash" Value="efa4a779d8ea642b0a5a70ab33e87fc742ffe3f7" /><Meta Name="ms.sourcegitcommit" Value="2767b5903996fbd7e2b81eda1114dfc9d0d9db63" /><Meta Name="ms.translationtype" Value="MT" /><Meta Name="ms.contentlocale" Value="fr-FR" /><Meta Name="ms.lasthandoff" Value="05/02/2019" /><Meta Name="ms.locfileid" Value="65014785" /></Metadata><TypeSignature Language="C#" Value="public class XamlInternalException : System.Xaml.XamlException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit XamlInternalException extends System.Xaml.XamlException" />
  <TypeSignature Language="DocId" Value="T:System.Xaml.XamlInternalException" />
  <TypeSignature Language="VB.NET" Value="Public Class XamlInternalException&#xA;Inherits XamlException" />
  <TypeSignature Language="C++ CLI" Value="public ref class XamlInternalException : System::Xaml::XamlException" />
  <TypeSignature Language="F#" Value="type XamlInternalException = class&#xA;    inherit XamlException" />
  <AssemblyInfo>
    <AssemblyName>System.Xaml</AssemblyName>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Xaml.XamlException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute FrameworkAlternate="netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8;netcore-3.0">
      <AttributeName>System.Serializable</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary><span data-ttu-id="cdc33-101">L’exception est levée en cas d’incohérences internes qui se produisent pendant la lecture de XAML et l’écriture de XAML.</span><span class="sxs-lookup"><span data-stu-id="cdc33-101">The exception that is thrown for internal inconsistencies that occur during XAML reading and XAML writing.</span></span></summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="cdc33-102">Cette exception est généralement levée uniquement lorsqu’un aspect du lecteur XAML ou du système de writer XAML a été étendu par votre code, et des incohérences ou des problèmes d’implémentation s’est produite qui ont été passés au code du système de Services XAML du .NET Framework.</span><span class="sxs-lookup"><span data-stu-id="cdc33-102">This exception is usually thrown only when some aspect of the XAML reader or XAML writer system was extended by your code, and inconsistencies or implementation problems occurred that were passed on to the .NET Framework XAML Services system code.</span></span> <span data-ttu-id="cdc33-103">Les exceptions qui peuvent provenir de tierces parties, telles que l’analyse XAML et les erreurs dans le XAML, ne doivent pas lever <xref:System.Xaml.XamlInternalException>, à la place lever d’autres exceptions comme <xref:System.Xaml.XamlParseException> ou <xref:System.Xaml.XamlObjectWriterException>.</span><span class="sxs-lookup"><span data-stu-id="cdc33-103">Exceptions that might originate from third parties, such as parsing XAML and encountering errors in the XAML, should not throw <xref:System.Xaml.XamlInternalException>, and should instead throw other exceptions such as <xref:System.Xaml.XamlParseException> or <xref:System.Xaml.XamlObjectWriterException>.</span></span>  
  
 <span data-ttu-id="cdc33-104">Si vous écrivez du code qui étend les Services XAML .NET Framework, en particulier, si vous écrivez du code pour un lecteur XAML ou d’une implémentation du writer XAML, vous devez conserver cette distinction entre <xref:System.Xaml.XamlInternalException> et les autres exceptions liées au XAML.</span><span class="sxs-lookup"><span data-stu-id="cdc33-104">If you are writing code that extends .NET Framework XAML Services, in particular, if you are writing code for a XAML reader or XAML writer implementation, you should maintain this distinction between <xref:System.Xaml.XamlInternalException> and the other XAML-related exceptions.</span></span> <span data-ttu-id="cdc33-105">Votre code doit uniquement lever <xref:System.Xaml.XamlInternalException> dans les cas où une extension plus poussée de votre classe ne respecte pas un modèle de conception principal ou d’implémentation qui définit votre code pouvant être appelé.</span><span class="sxs-lookup"><span data-stu-id="cdc33-105">Your code should only throw <xref:System.Xaml.XamlInternalException> in cases where further extension of your class violates some design principle or implementation pattern that your callable code defines.</span></span> <span data-ttu-id="cdc33-106">Votre code doit lever les autres exceptions spécifiques au XAML pour les cas d’exception où votre lecteur XAML ou l’implémentation du writer XAML est utilisée sans extension et est utilisée pour charger le XAML de tiers, le contexte de schéma XAML par des tiers et ainsi de suite.</span><span class="sxs-lookup"><span data-stu-id="cdc33-106">Your code should throw other XAML-specific exceptions for exception cases where your XAML reader or XAML writer implementation is used without extension and is used to load third-party XAML, third-party XAML schema context, and so on.</span></span>  
  
 ]]></format>
    </remarks>
    <altmember cref="T:System.Xaml.XamlParseException" />
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>System.Xaml</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary><span data-ttu-id="cdc33-107">Initialise une nouvelle instance de la classe <see cref="T:System.Xaml.XamlInternalException" />.</span><span class="sxs-lookup"><span data-stu-id="cdc33-107">Initializes a new instance of the <see cref="T:System.Xaml.XamlInternalException" /> class.</span></span></summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public XamlInternalException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Xaml.XamlInternalException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; XamlInternalException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Xaml</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary><span data-ttu-id="cdc33-108">Initialise une nouvelle instance de la classe <see cref="T:System.Xaml.XamlInternalException" /> avec un message système qui décrit l'erreur.</span><span class="sxs-lookup"><span data-stu-id="cdc33-108">Initializes a new instance of the <see cref="T:System.Xaml.XamlInternalException" /> class with a system-supplied message that describes the error.</span></span></summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public XamlInternalException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Xaml.XamlInternalException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; XamlInternalException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new System.Xaml.XamlInternalException : string -&gt; System.Xaml.XamlInternalException" Usage="new System.Xaml.XamlInternalException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Xaml</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="cdc33-109">Message décrivant l'exception.</span><span class="sxs-lookup"><span data-stu-id="cdc33-109">The message that describes the exception.</span></span></param>
        <summary><span data-ttu-id="cdc33-110">Initialise une nouvelle instance de la classe <see cref="T:System.Xaml.XamlInternalException" /> avec un message spécifié décrivant l'erreur.</span><span class="sxs-lookup"><span data-stu-id="cdc33-110">Initializes a new instance of the <see cref="T:System.Xaml.XamlInternalException" /> class with a specified message that describes the error.</span></span></summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected XamlInternalException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Xaml.XamlInternalException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; XamlInternalException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberSignature Language="F#" Value="new System.Xaml.XamlInternalException : System.Runtime.Serialization.SerializationInfo * System.Runtime.Serialization.StreamingContext -&gt; System.Xaml.XamlInternalException" Usage="new System.Xaml.XamlInternalException (info, context)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Xaml</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Runtime.TargetedPatchingOptOut("Performance critical to inline this type of method across NGen image boundaries")</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info"><span data-ttu-id="cdc33-111">Objet qui contient les données sérialisées de l'objet.</span><span class="sxs-lookup"><span data-stu-id="cdc33-111">The object that holds the serialized object data.</span></span></param>
        <param name="context"><span data-ttu-id="cdc33-112">Informations contextuelles sur la source ou la destination.</span><span class="sxs-lookup"><span data-stu-id="cdc33-112">The contextual information about the source or destination.</span></span></param>
        <summary><span data-ttu-id="cdc33-113">Initialise une nouvelle instance de la classe <see cref="T:System.Xaml.XamlInternalException" /> avec des données sérialisées.</span><span class="sxs-lookup"><span data-stu-id="cdc33-113">Initializes a new instance of the <see cref="T:System.Xaml.XamlInternalException" /> class with serialized data.</span></span></summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public XamlInternalException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Xaml.XamlInternalException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; XamlInternalException(System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new System.Xaml.XamlInternalException : string * Exception -&gt; System.Xaml.XamlInternalException" Usage="new System.Xaml.XamlInternalException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Xaml</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="cdc33-114">Message décrivant l'exception.</span><span class="sxs-lookup"><span data-stu-id="cdc33-114">The message that describes the exception.</span></span></param>
        <param name="innerException"><span data-ttu-id="cdc33-115">Exception ayant provoqué l'exception actuelle.</span><span class="sxs-lookup"><span data-stu-id="cdc33-115">The exception that is the cause of the current exception.</span></span> <span data-ttu-id="cdc33-116">Si le paramètre <paramref name="innerException" /> n'est pas <see langword="null" />, l'exception actuelle est levée dans un bloc <see langword="catch" /> qui gère l'exception interne.</span><span class="sxs-lookup"><span data-stu-id="cdc33-116">If the <paramref name="innerException" /> parameter is not <see langword="null" />, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span></span></param>
        <summary><span data-ttu-id="cdc33-117">Initialise une nouvelle instance de la classe <see cref="T:System.Xaml.XamlInternalException" /> avec un message d'erreur spécifié et une référence à l'exception interne ayant provoqué cette exception.</span><span class="sxs-lookup"><span data-stu-id="cdc33-117">Initializes a new instance of the <see cref="T:System.Xaml.XamlInternalException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span></span></summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
  </Members>
</Type>